src/authextension.ts

```diff
@@ -14,16 +15,23 @@ export module authextension {
     process.env['THREE_SCALE_USER_TOKEN'] = apiConfig.apiKey;
     process.env['PROVIDE_FULLSTACK_ACTION'] = 'true';
     process.env['GOLANG_EXECUTABLE'] = Config.getGoExecutable();
+    process.env['UTM_SOURCE'] = 'vs-code';
   };
 
   export function setUUID(uuid) {
     process.env['UUID'] = uuid;
   }
 
+  export async function getTelemetryid() {
+    process.env['TELEMETRY_ID'] = await getRedHatUUID();
+  }
+
   export const authorize_f8_analytics = async context => {
     try {
       setContextData(apiConfig);
 
+      getTelemetryid();
+
```

Comment:
Can this before setContextData and add await, otherwise there is no guarantee that the env changes would get propagated to lsp.

Reply:
@arajkumar It should get propagated as uuid propagation is also happening after this

Reply:
use await

Reply:
Done